lib/iris/_data_manager.py

```diff
@@ -308,8 +325,12 @@ def data(self):
                         'before trying again.')
                 raise MemoryError(emsg.format(self.shape, self.dtype))
 
-            # Check the manager contract, as the managed data has changed.
-            self._assert_axioms()
+        if ma.isMaskedArray(self._real_array):
+            # Align the numpy fill-value with the data manager fill-value.
+            self._real_array.fill_value = self.fill_value
+
+        # Check the manager contract, as the managed data has changed.
+        self._assert_axioms()
```

Comment:
@djkirkham Actually, after some thought, no, I disagree. 

We always want to do lines 328-333 regardless of whether the data manager `has_lazy_data` or not.